src/components/document-metadata-editor.js

```diff
@@ -1,181 +1,162 @@
-import React from 'react';
-import autoBind from 'auto-bind';
 import PropTypes from 'prop-types';
 import urls from '../utils/urls.js';
-import { inject } from './container-context.js';
-import { withTranslation } from 'react-i18next';
-import { withSettings } from './settings-context.js';
-import { withLanguage } from './language-context.js';
+import { useTranslation } from 'react-i18next';
+import React, { useEffect, useState } from 'react';
+import { useService } from './container-context.js';
+import { useSettings } from './settings-context.js';
+import { useLanguage } from './language-context.js';
 import { handleApiError } from '../ui/error-helper.js';
 import inputValidators from '../utils/input-validators.js';
 import { Input, Radio, Tag, Space, Select, Form } from 'antd';
 import { EyeOutlined, EditOutlined } from '@ant-design/icons';
 import LanguageSelect from './localization/language-select.js';
+import { documentRevisionShape } from '../ui/default-prop-types.js';
 import LanguageNameProvider from '../data/language-name-provider.js';
 import DocumentApiClient from '../api-clients/document-api-client.js';
 import CountryFlagAndName from './localization/country-flag-and-name.js';
-import { documentRevisionShape, translationProps, languageProps, settingsProps } from '../ui/default-prop-types.js';
 
 const RadioButton = Radio.Button;
 const RadioGroup = Radio.Group;
 
 const MODE_EDIT = 'edit';
 const MODE_PREVIEW = 'preview';
 
-class DocumentMetadataEditor extends React.Component {
-  constructor(props) {
-    super(props);
-    autoBind(this);
-    this.documentApiClient = props.documentApiClient;
-    this.state = {
-      mode: MODE_PREVIEW,
-      tagsValidationStatus: '',
-      slugValidationStatus: '',
-      tagSuggestions: []
-    };
-  }
+function DocumentMetadataEditor({ documentRevision, onChanged }) {
+  const settings = useSettings();
+  const { language } = useLanguage();
+  const { t } = useTranslation('documentMetadataEditor');
+  const documentApiClient = useService(DocumentApiClient);
+  const languageNameProvider = useService(LanguageNameProvider);
 
-  componentDidMount() {
-    this.validateMetadata(this.props.documentRevision);
-  }
+  const [mode, setMode] = useState(MODE_PREVIEW);
+  const [tagSuggestions, setTagSuggestions] = useState([]);
+  const [slugValidationStatus, setSlugValidationStatus] = useState('');
+  const [tagsValidationStatus, setTagsValidationStatus] = useState('');
 
-  validateMetadata(metadata) {
+  const validateMetadata = metadata => {
     const isValidSlug = inputValidators.isValidSlug(metadata.slug);
-    this.setState({ slugValidationStatus: isValidSlug ? '' : 'error' });
+    setSlugValidationStatus(isValidSlug ? '' : 'error');
```

Comment:
I think we should make this a warning, because we have a lot of docs where it would be red and you can still save them (unless you type something and make it valide)

Reply:
I'm wondering if we can really make it just a warning in any case, what if I just give you the following slug " "?
I think it could be made to be considered "valid" but also pointless or?

We would need to relax the server validation as well. I've tried locally and it doesn't work at the moment.

Reply:
Given the points brought up here, I opted for making a dedicated ticket for this (https://educandu.atlassian.net/browse/EDU-251) and keeping this PR as a change that should not have side-effects and behave as before.

Is that ok with you?